frontend/src/metabase/visualizations/echarts/cartesian/option/index.ts

```diff
@@ -87,14 +88,14 @@ export const getCartesianChartOption = (
   // dataset option
   const dimensions = [
     X_AXIS_DATA_KEY,
-    ...chartModel.seriesModels.map(seriesModel => seriesModel.dataKey),
-  ];
-
-  if (settings["stackable.stack_type"] != null) {
-    dimensions.push(
-      ...[POSITIVE_STACK_TOTAL_DATA_KEY, NEGATIVE_STACK_TOTAL_DATA_KEY],
-    );
-  }
+    POSITIVE_STACK_TOTAL_DATA_KEY,
+    NEGATIVE_STACK_TOTAL_DATA_KEY,
+    ...chartModel.seriesModels.map(seriesModel => [
+      seriesModel.dataKey,
+      getBarSeriesDataLabelKey(seriesModel.dataKey, "+"),
```

Comment:
We push these keys to the `dimensions` property, even for non-bar series, because there's no harm in having it, and it's more efficient to not filter for bar series. For non-bar series since there is no data under these data keys nothing will be rendered.